### PR TITLE
ENH: update `random.binomial`

### PR DESCRIPTION
**Pull request by samir-nasibli**
## Description
* left only dparray alloc and fptr call on cython layer
* update dpnp_rng_binomial_c

## Tests
tests/test_random.py::TestDistributionsBinomial::test_extreme_value                                                                                                                                                       
tests/test_random.py::TestDistributionsBinomial::test_invalid_args
tests/test_random.py::TestDistributionsBinomial::test_moments
tests/test_random.py::TestDistributionsBinomial::test_seed

### CODE REVIEW

**Comment by samir-nasibli** [comment]
If changes will be approved, all similar changes will be added on one other PR.

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 1]
```diff
@@ -114,21 +114,36 @@ void dpnp_rng_binomial_c(void* result, const int ntrial, const double p, const s
...
-    if (dpnp_queue_is_cpu_c())
+    if (ntrial == 0 || p == 0)
     {
-        mkl_rng::binomial<_DataType> distribution(ntrial, p);
-        // perform generation
-        auto event_out = mkl_rng::generate(distribution, DPNP_RNG_ENGINE, size, result1);
-        event_out.wait();
+        _DataType* fill_value = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(sizeof(_DataType)));
+        fill_value[0] = 0;
+        dpnp_initval_c<_DataType>(result, fill_value, size);
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:121]: That is similar to `dpnp_zeros_c<_DataType>(result, size);`

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 2]
```diff
@@ -114,21 +114,36 @@ void dpnp_rng_binomial_c(void* result, const int ntrial, const double p, const s
...
+    }
+    else if (p == 1)
+    {
+        _DataType* fill_value = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(sizeof(_DataType)));
+        fill_value[0] = static_cast<_DataType>(ntrial);
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:126]: I think we need to free `fill_value` after it's not used more.

Comment (samir-nasibli) [reply dpnp/backend/kernels/dpnp_krnl_random.cpp:126]: oh, yeah, of course. Thanks

File: dpnp/backend/kernels/dpnp_krnl_random.cpp (hunk 1 above)
Comment (samir-nasibli) [reply dpnp/backend/kernels/dpnp_krnl_random.cpp:121]: Yeah, this looks more preferable. I see that `dpnp_zeros_c` is already implemented. Good point.

**Review by densmirn — APPROVED** [review]

**Review by shssf — APPROVED** [review]

**Pull request merged**